projects/plugins/jetpack/extensions/blocks/conversation/edit.js

```diff
@@ -94,8 +112,100 @@ function ConversationEdit( { className, attributes, setAttributes } ) {
 		} );
 	}
 
+	function showTranscriptProcessErrorMessage( message ) {
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice( message );
+		setIsProcessingFile( false );
+	}
+
+	function uploadTranscriptFile( event ) {
+		const transcriptFile = event.target.files?.[ 0 ];
+
+		// Check file exists.
+		if ( ! transcriptFile ) {
+			return showTranscriptProcessErrorMessage( __( 'Transcript file not found.', 'jetpack' ) );
+		}
+
+		// Check file MAX size.
+		if (
+			transcriptFile?.size && transcriptFile.size <= 0 || // min size
+			! transcriptFile?.size || transcriptFile.size > TRANSCRIPT_MAX_FILE_SIZE // max size
+		) {
+			return showTranscriptProcessErrorMessage( __( 'Invalid transcript file size.', 'jetpack' ) );
+		}
+
+		// Check file type.
+		if ( transcriptFile?.type?.length && transcriptFile.type !== 'text/plain' ) {
+			return showTranscriptProcessErrorMessage( __( 'Invalid transcript file type.', 'jetpack' ) );
+		}
+
+		// Check format by extension.
+		const fileExtension = pickExtensionFromFileName( transcriptFile?.name );
+		if ( ! isAcceptedTranscriptExtension( fileExtension ) ) {
+			return showTranscriptProcessErrorMessage( __( 'Invalid transcript file extension.', 'jetpack' ) );
+		}
+
+		setIsProcessingFile( true );
+
+		parseTranscriptFile( transcriptFile, function( { conversation, dialogues }, err ) {
+			if ( err ) {
+				return showTranscriptProcessErrorMessage( err );
+			}
+
+			setAttributes( {
+				participants: conversation.speakers,
+				skipUpload: ! conversation?.length,
+			} );
+			const dialogueBlocksTemplate = dialogues.map( ( dialogue ) => [
+				'jetpack/dialogue',
+				dialogue,
+			] );
+
+			const dialogueBlocks = createBlocksFromInnerBlocksTemplate( dialogueBlocksTemplate );
+			insertBlocks( dialogueBlocks, 0, clientId );
+			setIsProcessingFile( false );
+		} );
+	}
+
 	const baseClassName = 'wp-block-jetpack-conversation';
 
+	if ( ! participants?.length && ! skipUpload ) {
+		return (
+			<Placeholder
+				label={ __( 'Conversation', 'jetpack' ) }
+				instructions={ __(
+					'Upload a transcript file or create a conversation with blank content.',
+					'jetpack'
+				) }
+				icon={ <BlockIcon icon={ icon } /> }
+				notices={ noticeUI }
+			>
+				<div className={ `${ baseClassName }__placeholder` }>
+					<FormFileUpload
+						multiple={ false }
+						isLarge
+						className="wp-block-jetpack-slideshow__add-item-button"
+						onChange={ uploadTranscriptFile }
+						accept={ ACCEPTED_FILE_EXTENSIONS }
+						isPrimary
+						title={ `${ __( 'Accepted file formats:', 'jetpack' ) } ${ ACCEPTED_FILE_EXTENSIONS }` }
```

Comment:
Maybe this should be one strong with a placeholder for the file formats?

Reply:
addressed here https://github.com/Automattic/jetpack/pull/19005/commits/3446106a9d9abc21942cac97d05a0b3cab705912